### ISSUE TITLE
Project server crashed - client messed the computer location.

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#633]
**Reported by Pepo on 20 Sep 38373543 21:20 UTC**
SETI's BOINC database server crashed today. Few minutes later my host contacted it:

```
19:48:15 [[sched_op_debug](SETI@home]) Starting scheduler request
19:48:15 [Sending scheduler request: To fetch work.  Requesting 8 seconds of work, reporting 0 completed tasks
19:48:20 [SETI@home](SETI@home]) Scheduler request succeeded: got 0 new tasks`
19:48:20 [[sched_ops_debug](SETI@home]) Server version 601
19:48:20 [Message from server: Server can't open database
19:48:20 [SETI@home](SETI@home]) Project requested delay of 3600.000000 seconds
19:48:20 [New computer location: 
19:48:20 [---](SETI@home]) General prefs: from SETI@home (last modified 13-May-2008 10:27:28)
19:48:20 [Host location: none
19:48:20 [---](---]) General prefs: using your defaults
19:48:20 [Reading preferences override file
19:48:20 [---](---]) Preferences limit memory usage when active to 1023.02MB
19:48:20 [Preferences limit memory usage when idle to 1841.44MB
19:48:20 [---](---]) Preferences limit disk usage to 0.00GB
```

sched_reply_SETI.xml contained just following:

```
<scheduler_reply>
<scheduler_version>601</scheduler_version>
<master_url>http://setiathome.berkeley.edu/</master_url>
<request_delay>3600.000000</request_delay>
<message priority="low">Server can't open database </message>
<project_name>SETI@home</project_name>
</scheduler_reply> 
```

The host was "thrown off" from its venue and related settings and the mess started: the machine was constantly suspending/resuming CPU throttling, switched off network and effectively nothing was being crunched anymore.

The client should be resistent against such empty returned data.

**Comment by romw** [comment]
**Commented by davea on 17 Mar 38373633 01:46 UTC**
(In [15230]) - client: If a scheduler reply doesn't include <host_venue>
    (as is the case if its DB is down,
    or it has no work and skips the DB check)
    don't overwrite the host_venue on the client.
    Fixes #656, I hope (Rom: please port)

**Comment by romw** [comment]
**Commented by Nicolas on 4 Feb 38373677 04:00 UTC**
So the fix in #578 was useless, because the client still deleted the venue even if the server returned <project_is_down/>?

Also, there is an identical problem with team name, see #417 and #633.
